doc/src/devdocs/offset-arrays.md

```diff
@@ -75,20 +76,12 @@ can sometimes simplify such tests.
 
 ### Linear indexing (`linearindices`)
 
-Some algorithms are most conveniently (or efficiently) written in terms of a single linear index,
-`A[i]` even if `A` is multi-dimensional.  In "true" linear indexing, the indices always range
-from `1:length(A)`. However, this raises an ambiguity for one-dimensional arrays (a.k.a., [`AbstractVector`](@ref)):
-does `v[i]` mean linear indexing, or Cartesian indexing with the array's native indices?
 
-For this reason, if you want to use linear indexing in an algorithm, your best option is to get
-the index range by calling `linearindices(A)`.  This will return `indices(A, 1)` if `A` is an
-`AbstractVector`, and the equivalent of `1:length(A)` otherwise.
+Some algorithms are most conveniently (or efficiently) written in terms of a single linear index, `A[i]` even if `A` is multi-dimensional. Regardless of the array's native indices, linear indices always range from `1:length(A)`. However, this raises an ambiguity for one-dimensional arrays (a.k.a., [`AbstractVector`](@ref)): does `v[i]` mean linear indexing , or Cartesian indexing with the array's native indices?
 
-In a sense, one can say that 1-dimensional arrays always use Cartesian indexing. To help enforce
-this, it's worth noting that `sub2ind(shape, i...)` and `ind2sub(shape, ind)` will throw an error
-if `shape` indicates a 1-dimensional array with unconventional indexing (i.e., is a `Tuple{UnitRange}`
-rather than a tuple of `OneTo`).  For arrays with conventional indexing, these functions continue
-to work the same as always.
+For this reason, if you want to use linear indexing in an algorithm, your best option is to iterate over the array with `eachindex(A)`, or, if you require the indices to be sequential integers, to get the index range by calling `linearindices(A)`. This will return `indices(A, 1)` if A is an AbstractVector, and the equivalent of `1:length(A)` otherwise.
```

Comment:
Note this starts with "use linear indexing" but then recommends `eachindex` first (which may not yield a linear index). Maybe better as
```
For this reason, your best option may be to iterate over the array with `eachindex(A)`, or, if you require...
```

Reply:
good point, fixed